### DISCUSSION TITLE
Provide checksums/signed binaries for releases

### DISCUSSION THREAD

**Discussion by infinitis** [ppy/osu#19123] [Ideas]
Would love to be able to verify release binaries with signed checksums if possible (looks like the RELEASES are only for the nuget packages?).

**Comment by peppy** [comment]
Can you explain your use case? This would generally be managed automatically by the updater.

**Comment by infinitis** [comment]
Security reasons. Downloading and running untrusted binaries is a massive security hole. Currently, as a user I have no way to guarantee I'm running exactly what is uploaded by the devs.

**Reply by peppy** [reply]
This confuses me. If someone is able to replace a binary on github releases, they would also be able to change a hash in the attached release description.

I think if you are worried about github security, then your best bet is to build from source.

**Comment by infinitis** [comment]
Hence the signed part of "signed checksum". It's fairly common for open
source projects and personally am not a fan of running software
without such attestation.

It my understanding (in at least debian, for instance) that signing
packages is standard in distribution of binaries/packages for open
source projects.

And as much as I love playing osu, I'm not a fan of building from source
for this particular project for a couple of different reasons.

**Reply by peppy** [reply]
No worries. You can wait for app image updater to come online or use a package manager that distributes our binaries. I don't see any benefit in publishing checksums to github as it does not add any security benefits from my perspective.

**Reply by Soetch** [reply]
Real, if it does add a little more "official" feeling, it's not that important (to my eyes). I think you can trust this repository enough to build from its source.
